worker/session.go

```diff
@@ -229,16 +230,25 @@ func (sp *sessionPool) acquire(ctx context.Context, s *session) (_ *sharedSessio
 	}
 
 reconnect:
+	if ss.sess != nil && sp.sessionReconnectTimeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, sp.sessionReconnectTimeout)
```

Comment:
I'd use a different name for the context here. Otherwise we end up with that timeout not just when dialing the host but pass it into all the followings functions using it without realising.

Reply:
I kind of want it to be passed into all following functions that deal with (re)connecting a host. The timeout should apply to the reconnect as a whole, not just on the host dial. Right?